cdap-ai-spi/src/main/java/io/cdap/cdap/ai/spi/AIProvider.java

```diff
@@ -32,9 +32,9 @@ default void initialize(AIProviderContext context) throws Exception {
      /**
       * Summarizes the given application details in mentioned format.
       *
-      * @param applicationDetail The detailed information about the application.
+      * @param appRequest The detailed information about the application.
       * @param format The format in which the summary should be returned.
       * @return A summarized representation of the application details, formatted according to the specified format.
       */
-     public String summarizeApp(ApplicationDetail applicationDetail, String format);
+     public String summarizeApp(AppRequest appRequest, String format);
```

Comment:
What is the value of `format` we support? Using a string seems too free formed. Consider using Enum.